cmd/api/routers/routers.go

```diff
@@ -47,6 +47,30 @@ func (c *Controller) GetNamespacedResources(context *gin.Context) {
 	context.JSON(http.StatusOK, resources)
 }
 
+func (c *Controller) GetAllCoreResources(context *gin.Context) {
+	version := context.Param("version")
+	kind := getAPIResourceKind(context)
+	resources, err := c.Database.QueryCoreResources(context.Request.Context(), kind, version)
+
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, err.Error())
+	}
+	context.JSON(http.StatusOK, resources)
```

Comment:
Where can I find documentation about this usage? Specifically, why the second call to `context.JSON` is not in an `else`.

Reply:
In their docs: https://pkg.go.dev/github.com/gin-gonic/gin#Context.JSON

And I think you are right,  that context.JSON needs to be `abort.JSON` if it needs to exit. We have a small utility function for that. And this sounds like we are not covering the return code in a test, or just the OK case.

Reply:
I've updated this to the abort so that we return after setting the JSON output to an error

Reply:
We needed also the return, if not, several JSONs are appended to the response.

I've also added some tests for checking the output on failure